src/nsrt_learning/strips_learning/oracle_learner.py

```diff
@@ -27,7 +28,18 @@ def _learn(self) -> List[PartialNSRTAndDatastore]:
             pnads.append(
                 PartialNSRTAndDatastore(nsrt.op, datastore, option_spec))
         self._recompute_datastores_from_segments(pnads)
-        return pnads
+        # Filter out any pnad that has an empty datastore. This can occur when
+        # using non-standard settings with environments that cause certain
+        # operators to be unnecessary. For example, in painting, when using
+        # --painting_goal_receptacles box, the operator for picking from
```

Comment:
better to enclose in backquotes for readability: `--painting_goal_receptacles box`